### ISSUE TITLE
need to be able to share url's and specify organism

### ISSUE THREAD

**Issue by nathandunn** [GMOD/Apollo#219]
1 - need to be able to share url's 

Can not copy and paste URL's and use them separately . . . need to be able to do that and specify the organism.

**Comment by nathandunn** [comment]
can also "generate" a URL, as well

**Comment by nathandunn** [comment]
It is in the link icon . . .

**Comment by nathandunn** [comment]
link should be in the blue section to *ALWAYS* represent what is going on in the JBRowse windwo

**Comment by nathandunn** [comment]
can remove "full-screen"

**Comment by nathandunn** [comment]
Should support URL's that look like this (note added "organism"):

    http://localhost:8080/apollo/annotator/index?loc=Group1.3%3A445831..797889&highlight=&tracklist=1&tracks=DNA%2CAnnotations%2COfficial%20Gene%20Set%20v3.2&nav=1&overview=1&organism=7

Conversely, URL's that look like this:

http://localhost:8080/apollo/jbrowse/?loc=Group1.3%3A445831..797889&highlight=&tracklist=1&tracks=DNA%2CAnnotations%2COfficial%20Gene%20Set%20v3.2&nav=1&overview=1&organism=7

Should have links out back to the top one with the full URL.  Adding the organism will be interesting.  Grabbing the URL from the session should be pretty easy.  However, propagating onto the URL may be difficult.

**Comment by nathandunn** [comment]
We could either Plan A:
1) shift it so we pass in a single parameter that maps the entire session after doing a redirect via a controller and then you just use it normally.    
2) if you're not logged in then redirect to the annotator page.  If folks want to click on the full URL, they can.  
3) we use our same link still from the GWT side, but we grab organization, sequence, and track info, and location data from the frame (I think that is what we'd done before?)
4) if we pass in organization, sequence, locations, and track info into the annotator/index it should be able to parse that data and set our session appropriately (once we've logged in)
5) can our "full-view / JBrowse view" also grab the organization out as part of the url from the session?    On login, it could return with that and the session would be set per 1) above.
6) in the jbrowse-view, we should add a separate option to view to pass in the "annotator" URL (just the simple one).

**Comment by monicacecilia** [comment]
@nathandunn While I am able to retrieve a URL and open the same region in a different browser, I am receiving a number of errors in doing so. The error report reads as follows: 
<code> 
Uncaught com.google.gwt.json.client.JSONException: Error parsing JSON: SyntaxError: Unexpected token
</code>

Also, two curious things of note: 
1. In Chrome, I was not able to simply highlight and use keyboard commands to copy the text that appears in the box (after clicking on the link button). I had to either use the "Edit/Copy" menu, or use the right click menu to get the URL. 
2. The second browser (Safari in this case) took a very long time in retrieving the annotation I was seeing in the first browser (Chrome).

**Comment by nathandunn** [comment]
Also, two curious things of note: 
1. In Chrome, I was not able to simply highlight and use keyboard commands to copy the text that appears in the box (after clicking on the link button). I had to either use the "Edit/Copy" menu, or use the right click menu to get the URL. 

@monicacecilia   I agree that is annoying.  I could slide it under the link / above the tabs (moving them all down)

2. The second browser (Safari in this case) took a very long time in retrieving the annotation I was seeing in the first browser (Chrome).

@monicacecilia  Can you provide the link for me?  Not sure what is going on there.

**Comment by monicacecilia** [comment]
@nathandunn 
http://icebox.lbl.gov/Apollo2/annotator/loadLink?loc=Group1.32:402680..421500&organism=5649&highlight=0&tracklist=0&tracks=Official Gene Set v3.2

**Comment by nathandunn** [comment]
TODO (for me):
1 - encode link (location & tracks)
2 - use Bootstrap Dialog (preferred) to see if we can't use Command-C to copy. - OR - slide down a vertical panel below.

**Comment by nathandunn** [comment]
@monicacecilia   You'll have to let me know if this works UI-wise.   Popup's will not allow copy-paste to work from the terminal (that is just an internetty thing, though if you know a workaround . . .. ).   I think we can get pretty flexible with what we end up showing and how things work.

**Comment by monicacecilia** [comment]
It works almost perfectly, and then there's a twist.
I was able to open the URLs on Firefox and safari -- after login it went straight to the place I posted.  - at that zoom level. BUT, if I try to open a second window on the same browser using that same URL, the browser no longer goes down to the coordinates, instead it just displays the entire scaffold/group "zoomed all the way out". This is bizarre behaviour --  I'd like to know what's behind it.  
(It also happens on Chrome, so it is not a browser-specific thing. this is on the Apollo side).

For example: the window above is Firefox the first time I opened the window. The second window is shown below that one.
![screen shot 2015-05-21 at 5 51 02 pm](https://cloud.githubusercontent.com/assets/1714621/7761967/04e427cc-ffe2-11e4-95d0-16c0e48a4d39.png)

I used this URL, but it has done that with each URL. 
http://icebox.lbl.gov/Apollo2/annotator/loadLink?loc=Group1.17:71771..645048&organism=4&highlight=0&tracklist=0&tracks=Reference sequence,Official Gene Set v3.2,User-created Annotations,Nurse RNA-Seq reads,Nurse RNA-Seq XY Plot

almost ready for :fireworks:

**Comment by nathandunn** [comment]
The URL should have been (when I clicked on that transcript): 

http://icebox.lbl.gov/Apollo2/annotator/loadLink?loc=Group1.17:416195..431195&organism=4&highlight=0&tracklist=0&tracks=Reference sequence,User-created Annotations

It is getting the preference from the database.  If you happen to happen to have opened up the latter, it will open up all the way . .. that being said, I think that if you reset the preference it should reset that preference.  Let me see if I can't fix that.

**Comment by nathandunn** [comment]
Hmm . . . looking closer . .  I'm having trouble reproducing this.    Even if its set on the database, it only looks at the local browser to generate those coordinates.   Also, these aren't the full sequence, which are the coordinates it will start / stop coordinates are not set.

**Comment by nathandunn** [comment]
You can show me this when we you get in.

**Comment by nathandunn** [comment]
of course, then I update the server and now I see it.   Once I click on the transcript it is fine.  However, just navigating, it doesn't seem to be set internally.

**Comment by nathandunn** [comment]
Should be fixed!

**Comment by monicacecilia** [comment]
Just because at this juncture, this is just funny... 
I took the URL from Chrome (top), pasted in Firefox (bottom left) and Safari. ... two of these things are not like the Chrome. . . :stuck_out_tongue_winking_eye:
![screen shot 2015-05-22 at 2 29 42 pm](https://cloud.githubusercontent.com/assets/1714621/7780066/49e2b53e-008f-11e5-8da7-981d3a7c125d.png)

... then I refreshed it all, and it all works, and the URL sends user to the right place every time. hooray! :dancer:
